### ISSUE TITLE
Add X and Y axis labels for graphs

### ISSUE THREAD

**Issue by eah13** [txst-per-group/Glowscript-Blocks#110]
Glowscript 2.6 supports better X and Y axis labels.  Blocks could too.

Discussion needed on the best way to do this.

Example:
https://trinket.io/glowscript/86cb69308c

**Comment by codestar12** [comment]
It seems that if you include a title and xaxis, taxis it doesn't interfere with with anything. My suggestion then would be to just add those blocks to the mutator and generator and leave title alone. This way we don't break any old XML programs using title

**Comment by dube1503** [comment]
A question I have is what if we left title alone, but then on mutation instead of adding a 'title' input on the block we added a 'x axis' and 'y axis' inputs on the block in its place. While this would break old XML that include title, it would seem more intuitive i think then adding more mutator stubs in the menu that start with 'x' and 'y' as i think that could get confusing as well

**Comment by dube1503** [comment]
![img_0207](https://user-images.githubusercontent.com/19582167/30112390-d89b4364-92d6-11e7-8dc4-ee90c9d36813.JPG)

**Comment by eah13** [comment]
@dube1503 one thing that plan doesn't seem to contemplate is the different purposes of titles vs axis labels (if I understand you correctly).  Seems like one could want one without the other: A titled, un-axis-labeled graph, and untitled, axis labeled graph, or a titled and axis labeled graph.

**Comment by dube1503** [comment]
I have added the xtitle and ytitle attributes to the graph display block and it only accepts strings and properly outputs the python code.
![screen shot 2017-11-10 at 10 43 22 am](https://user-images.githubusercontent.com/19582167/32668764-1e43edb6-c604-11e7-8f3d-301d0bfe974e.png)
![screen shot 2017-11-10 at 10 44 08 am](https://user-images.githubusercontent.com/19582167/32668766-1f88d646-c604-11e7-8326-a5b60744f0c1.png)
However, within the our local test environment it does not seem to render the xtitle and ytitle along the axis lines as they do in the example @eah13 posted. Perhaps this is version issue with our local environment?
![screen shot 2017-11-10 at 10 46 01 am](https://user-images.githubusercontent.com/19582167/32668868-5f7cfd4a-c604-11e7-991c-a675957b7843.png)

**Comment by codestar12** [comment]
@dube1503 I wouldn't worry about it. Like @eah13 this is enabled in the new version of trinkets. I think we will just have to merge to the develop branch to do proper testing. Additionally, we could see about updating the version of glowscript we use in our local environment.

**Comment by dube1503** [comment]
@codestar12 @eah13 @hgclose same question here as #112 , does this mutator menu need defaults that mirror the shapes mutator menu?

**Comment by eah13** [comment]
I think this is actually a better candidate for defaults than the print_inline block, since users will be much more likely to use the defaults for convenience.  The print block can take a range of blocks as input, which means the default may be useless for users who aren't printing text.

See also [those comments](https://github.com/txst-per-group/Glowscript-Blocks/issues/112#issuecomment-355359557)

So, in short, yes this seems like a good place to try to do what you mean and add a default block.  Ideally that block inserts syntactically valid code so in addition to the convenience factor it avoids making half-formed code.

**Comment by dube1503** [comment]
Sounds like we are on the same page on all points. I will start with the default blocks and keep them syntactically correct as well mirror the actual defaults for the inputs.
The only place where I can see a problem is within the xmax and ymax as I know that the "defaults" for these is not actually set and when unassigned the graph grows to fit the content, but blank graphs display a 1.0 max on both axis so my guess would be to start there.

**Comment by dube1503** [comment]
![screen shot 2018-01-05 at 4 00 03 pm](https://user-images.githubusercontent.com/19582167/34630413-989af038-f231-11e7-9b56-cb27fe9b8e87.png)
Graph block fully mutated with default inputs. If there are no changes to the wording or selective defaults for each input then I will go forward with the PR for this as well.

**Comment by eah13** [comment]
LGTM.
